builtin/provisioners/chef/resource_provisioner.go

```diff
@@ -745,6 +805,9 @@ func decodeConfig(d *schema.ResourceData) (*provisioner, error) {
 		UserName:              d.Get("user_name").(string),
 		UserKey:               d.Get("user_key").(string),
 		Version:               d.Get("version").(string),
+		RetryOnExitCode:       getIntList(d.Get("retry_on_exit_code")),
+		WaitForRetry:          d.Get("wait_for_retry").(int),
+		MaxRetries:            d.Get("max_retries").(int),
```

Comment:
Same here, the list is ordered alphabetically so these new ones should be ordered like that as well. But let me add a commit to tweak your work just a little to see if we can get to a solution that is inline with the rest of the code in this provisioner...